licence-checker/licence-checker.py

```diff
@@ -381,6 +389,13 @@ def main():
 
     config.licence = LicenceHeader(parsed_config['licence'])
     config.exclude_paths = set(parsed_config['exclude_paths'])
+    # Check whether we should use regex matching or full string matching.
+    if 'match_regex' in parsed_config:
+        config.match_regex = parsed_config['match_regex'].lower() in ("yes",
+                                                                      "true",
+                                                                      "t", "1")
+    else:
+        config.match_regex = False
```

Comment:
For this behaviour, the following takes fewer lines (and fewer spaces!):
```python3
    match_regex = parsed_config.get('match_regex', '0')
    config.match_regex = match_regex in ['yes', 'true', 't', '1']
```

But I think this would be better with some error checking. Maybe this?
```python3
    match_regex = parsed_config.get('match_regex', 'false')
    if match_regex not in ['true', 'false']:
        print('Invalid value for match_regex: {!r}. '
              'Should be "true" or "false".'
              .format(match_regex))
    config.match_regex = match_regex == 'true'
```

In particular, I think we should raise an error rather than silently ignoring `match_regex: "cabbage"` !